contracts/margin/external/BucketLender/EthWrapperForBucketLender.sol

```diff
@@ -93,4 +102,46 @@ contract EthWrapperForBucketLender
         // deposit the tokens
         return BucketLender(bucketLender).deposit(beneficiary, amount);
     }
+
+    /**
+     * Allows users to send eth directly to this contract and have it be wrapped and sent to a
+     * BucketLender to be lent for some margin position.
+     *
+     * @param  bucketLender  The address of the BucketLender contract to deposit money into
+     * @return               The bucket number that was deposited into
+     */
+    function withdrawEth(
+        address bucketLender,
+        uint256[] buckets,
+        uint256[] maxWeights
+    )
+        external
+        returns (uint256, uint256)
+    {
+        address owedToken = BucketLender(bucketLender).OWED_TOKEN();
+        address heldToken = BucketLender(bucketLender).HELD_TOKEN();
+        require(
+            owedToken == WETH,
+            "EthWrapperForBucketLender: Cannot withdraw from non-WETH BucketLender"
+        );
+
+        // withdraw the weth
+        (
+            uint256 owedTokenAmount,
+            uint256 heldTokenAmount
+        ) = BucketLender(bucketLender).withdraw(
+            buckets,
+            maxWeights,
+            msg.sender
+        );
+
+        // send all eth to msg.sender
+        WETH9(owedToken).withdraw(owedTokenAmount);
```

Comment:
Do we want to wrap this in an `if (owedTokenAmount > 0)`? What happens if you withdraw / transfer 0?